# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1365834416 -7200
#      Sat Apr 13 08:26:56 2013 +0200
# Node ID e22604d4bf8569481233399fbf1a2da75ff7ecf6
# Parent  14045b872a3d63fa6960847c9b4e520f85d34612
* files-dock-widget.cc(constructor): initial dir is where octave was started

diff --git a/libgui/src/files-dock-widget.cc b/libgui/src/files-dock-widget.cc
--- a/libgui/src/files-dock-widget.cc
+++ b/libgui/src/files-dock-widget.cc
@@ -99,22 +99,22 @@ files_dock_widget::files_dock_widget (QW
            SLOT (change_directory_up ()));
   connect (_sync_octave_directory_action, SIGNAL (triggered ()), this,
            SLOT (do_sync_octave_directory ()));
   connect (_sync_browser_directory_action, SIGNAL (triggered ()), this,
            SLOT (do_sync_browser_directory ()));
 
   // TODO: Add other buttons for creating directories
 
-  // Create the QFileSystemModel starting in the home directory
-  QString homePath = QDir::homePath ();
-
+  // Create the QFileSystemModel starting in the actual directory
+  QDir curr_dir;
   _file_system_model = new QFileSystemModel (this);
   _file_system_model->setFilter (QDir::NoDotAndDotDot | QDir::AllEntries);
-  QModelIndex rootPathIndex = _file_system_model->setRootPath (homePath);
+  QModelIndex rootPathIndex = _file_system_model->setRootPath (
+                                                  curr_dir.absolutePath ());
 
   // Attach the model to the QTreeView and set the root index
   _file_tree_view = new QTreeView (container);
   _file_tree_view->setModel (_file_system_model);
   _file_tree_view->setRootIndex (rootPathIndex);
   _file_tree_view->setSortingEnabled (true);
   _file_tree_view->setAlternatingRowColors (true);
   _file_tree_view->setAnimated (true);
